### ISSUE TITLE
Can't install homebridge on Raspberry Pi

### ISSUE THREAD

**Issue by sdorilas** [nfarina/homebridge#1174]
pi@raspberrypi:~ $ sudo npm install -g --unsafe-perm homebridge
/opt/node/bin/homebridge -> /opt/node/lib/node_modules/homebridge/bin/homebridge

> mdns@2.3.3 install /opt/node/lib/node_modules/homebridge/node_modules/mdns
> node-gyp rebuild

make: Entering directory '/opt/node/lib/node_modules/homebridge/node_modules/mdns/build'
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_sd.o
In file included from ../src/dns_sd.cpp:1:0:
../src/mdns.hpp:32:20: fatal error: dns_sd.h: No such file or directory
 #include <dns_sd.h>
                    ^
compilation terminated.
dns_sd_bindings.target.mk:149: recipe for target 'Release/obj.target/dns_sd_bindings/src/dns_sd.o' failed
make: *** [Release/obj.target/dns_sd_bindings/src/dns_sd.o] Error 1
make: Leaving directory '/opt/node/lib/node_modules/homebridge/node_modules/mdns/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 4.4.48+
gyp ERR! command "/opt/node/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/node/lib/node_modules/homebridge/node_modules/mdns
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.5.0
gyp ERR! not ok 
/opt/node/lib
└── (empty)

npm ERR! Linux 4.4.48+
npm ERR! argv "/opt/node/bin/node" "/usr/local/bin/npm" "install" "-g" "--unsafe-perm" "homebridge"
npm ERR! node v6.9.5
npm ERR! npm  v4.3.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1

npm ERR! mdns@2.3.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mdns@2.3.3 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mdns package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mdns
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mdns
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /root/.npm/_logs/2017-02-28T23_56_36_844Z-debug.log

**Comment by sdorilas** [comment]
Sudo npm install -g --unsafe-perm homebridge
/opt/node/bin/homebridge -> /opt/node/lib/node_modules/homebridge/bin/homebridge

> mdns@2.3.3 install /opt/node/lib/node_modules/homebridge/node_modules/mdns
> node-gyp rebuild

make: Entering directory '/opt/node/lib/node_modules/homebridge/node_modules/mdns/build'
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_sd.o
In file included from ../src/dns_sd.cpp:1:0:
../src/mdns.hpp:32:20: fatal error: dns_sd.h: No such file or directory
 #include <dns_sd.h>
                    ^
compilation terminated.
dns_sd_bindings.target.mk:149: recipe for target 'Release/obj.target/dns_sd_bindings/src/dns_sd.o' failed
make: *** [Release/obj.target/dns_sd_bindings/src/dns_sd.o] Error 1
make: Leaving directory '/opt/node/lib/node_modules/homebridge/node_modules/mdns/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 4.4.48+
gyp ERR! command "/opt/node/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/node/lib/node_modules/homebridge/node_modules/mdns
gyp ERR! node -v v6.9.5
gyp ERR! node-gyp -v v3.5.0
gyp ERR! not ok 
/opt/node/lib
└── (empty)

npm ERR! Linux 4.4.48+
npm ERR! argv "/opt/node/bin/node" "/usr/local/bin/npm" "install" "-g" "--unsafe-perm" "homebridge"
npm ERR! node v6.9.5
npm ERR! npm  v4.3.0
npm ERR! code ELIFECYCLE
npm ERR! errno 1

npm ERR! mdns@2.3.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mdns@2.3.3 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mdns package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mdns
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mdns
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /root/.npm/_logs/2017-03-01T03_21_23_699Z-debug.log

**Comment by Neelrp12** [comment]
I recommend using this website to install homebridge.
https://blog.bram.co.nl/howto-homebridge-on-raspberry-pi-3-for-ios-and-smart-devices/

**Comment by sdorilas** [comment]
Thank you for your help @Neelrp12. I think I was able to install it but I still cannot run homebridge. See below:

pi@raspberrypi:~ $ sudo npm install -g --unsafe-perm homebridge
/opt/node/bin/homebridge -> /opt/node/lib/node_modules/homebridge/bin/homebridge
/opt/node/lib
└── homebridge@0.4.16 

pi@raspberrypi:~ $ homebridge
bash: homebridge: command not found
pi@raspberrypi:~ $

**Comment by sdorilas** [comment]
Any ideas by anyone? Thanks!

**Comment by NorthernMan54** [comment]
I would suggest starting again, and following the instructions on the Wiki for installation on a RaspberryPI.  

https://github.com/nfarina/homebridge/wiki/Running-HomeBridge-on-a-Raspberry-Pi

These are the most comprehensive.

Based on your original posting it appears that you are missing dependancies for mdns, likely this one

sudo apt-get install libavahi-compat-libdnssd-dev

**Comment by sdorilas** [comment]
I started over and followed the guide above. I now get this:

pi@raspberrypi:~ $ npm install -g node-gyp
npm ERR! Linux 4.4.50+
npm ERR! argv "/opt/node/bin/node" "/usr/bin/npm" "install" "-g" "node-gyp"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! path /opt/node/lib/node_modules/node-gyp/node_modules/.bin
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall rmdir

npm ERR! Error: EACCES: permission denied, rmdir '/opt/node/lib/node_modules/node-gyp/node_modules/.bin'
npm ERR!     at Error (native)
npm ERR!  { Error: EACCES: permission denied, rmdir '/opt/node/lib/node_modules/node-gyp/node_modules/.bin'
npm ERR!     at Error (native)
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'rmdir',
npm ERR!   path: '/opt/node/lib/node_modules/node-gyp/node_modules/.bin' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/pi/npm-debug.log
pi@I :~ $ sudo npm install -g node-gyp
/opt/node/bin/node-gyp -> /opt/node/lib/node_modules/node-gyp/bin/node-gyp.js
/opt/node/lib
��������� node-gyp@3.5.0 

pi@raspberrypi:~ $ sudo node-gyp BUILDTYPE=Release rebuild
sudo: node-gyp: command not found
pi@raspberrypi:~ $ sudo npm

**Comment by NorthernMan54** [comment]
Try this

sudo npm install -g homebridge

PS Sorry that wiki page is a bit of mess, as it mixes together several different methods of install.  I didn't realize it was that hard to follow until I looked at it again.

**Comment by sdorilas** [comment]
We're getting somewhere. I was able to try it and it appeared to install but still:

pi@raspberrypi:~ $ sudo npm install -g homebridge
/opt/node/bin/homebridge -> /opt/node/lib/node_modules/homebridge/bin/homebridge
- abbrev@1.1.0 node_modules/homebridge/node_modules/abbrev
- aproba@1.1.1 node_modules/homebridge/node_modules/aproba
- asn1@0.2.3 node_modules/homebridge/node_modules/asn1
- assert-plus@0.2.0 node_modules/homebridge/node_modules/assert-plus
- asynckit@0.4.0 node_modules/homebridge/node_modules/asynckit
- aws-sign2@0.6.0 node_modules/homebridge/node_modules/aws-sign2
- aws4@1.6.0 node_modules/homebridge/node_modules/aws4
- balanced-match@0.4.2 node_modules/homebridge/node_modules/balanced-match
- caseless@0.12.0 node_modules/homebridge/node_modules/caseless
- co@4.6.0 node_modules/homebridge/node_modules/co
- code-point-at@1.1.0 node_modules/homebridge/node_modules/code-point-at
- concat-map@0.0.1 node_modules/homebridge/node_modules/concat-map
- brace-expansion@1.1.6 node_modules/homebridge/node_modules/brace-expansion
- console-control-strings@1.1.0 node_modules/homebridge/node_modules/console-control-strings
- core-util-is@1.0.2 node_modules/homebridge/node_modules/core-util-is
- assert-plus@1.0.0 node_modules/homebridge/node_modules/dashdash/node_modules/assert-plus
- dashdash@1.14.1 node_modules/homebridge/node_modules/dashdash
- deep-extend@0.4.1 node_modules/homebridge/node_modules/deep-extend
- delayed-stream@1.0.0 node_modules/homebridge/node_modules/delayed-stream
- combined-stream@1.0.5 node_modules/homebridge/node_modules/combined-stream
- delegates@1.0.0 node_modules/homebridge/node_modules/delegates
- extend@3.0.0 node_modules/homebridge/node_modules/extend
- extsprintf@1.0.2 node_modules/homebridge/node_modules/extsprintf
- forever-agent@0.6.1 node_modules/homebridge/node_modules/forever-agent
- fs.realpath@1.0.0 node_modules/homebridge/node_modules/fs.realpath
- assert-plus@1.0.0 node_modules/homebridge/node_modules/getpass/node_modules/assert-plus
- getpass@0.1.6 node_modules/homebridge/node_modules/getpass
- graceful-fs@4.1.11 node_modules/homebridge/node_modules/graceful-fs
- har-schema@1.0.5 node_modules/homebridge/node_modules/har-schema
- has-unicode@2.0.1 node_modules/homebridge/node_modules/has-unicode
- hoek@2.16.3 node_modules/homebridge/node_modules/hoek
- boom@2.10.1 node_modules/homebridge/node_modules/boom
- cryptiles@2.0.5 node_modules/homebridge/node_modules/cryptiles
- inherits@2.0.3 node_modules/homebridge/node_modules/inherits
- block-stream@0.0.9 node_modules/homebridge/node_modules/block-stream
- ini@1.3.4 node_modules/homebridge/node_modules/ini
- is-typedarray@1.0.0 node_modules/homebridge/node_modules/is-typedarray
- isarray@1.0.0 node_modules/homebridge/node_modules/isarray
- isstream@0.1.2 node_modules/homebridge/node_modules/isstream
- jsbn@0.1.1 node_modules/homebridge/node_modules/jsbn
- ecc-jsbn@0.1.1 node_modules/homebridge/node_modules/ecc-jsbn
- jodid25519@1.0.2 node_modules/homebridge/node_modules/jodid25519
- json-schema@0.2.3 node_modules/homebridge/node_modules/json-schema
- json-stringify-safe@5.0.1 node_modules/homebridge/node_modules/json-stringify-safe
- jsonify@0.0.0 node_modules/homebridge/node_modules/jsonify
- json-stable-stringify@1.0.1 node_modules/homebridge/node_modules/json-stable-stringify
- ajv@4.11.4 node_modules/homebridge/node_modules/ajv
- har-validator@4.2.1 node_modules/homebridge/node_modules/har-validator
- mime-db@1.26.0 node_modules/homebridge/node_modules/mime-db
- mime-types@2.1.14 node_modules/homebridge/node_modules/mime-types
- form-data@2.1.2 node_modules/homebridge/node_modules/form-data
- minimatch@3.0.3 node_modules/homebridge/node_modules/minimatch
- minimist@0.0.8 node_modules/homebridge/node_modules/minimist
- mkdirp@0.5.1 node_modules/homebridge/node_modules/fstream/node_modules/mkdirp
- mkdirp@0.5.1 node_modules/homebridge/node_modules/node-pre-gyp/node_modules/mkdirp
- semver@5.3.0 node_modules/homebridge/node_modules/node-pre-gyp/node_modules/semver
- nopt@3.0.6 node_modules/homebridge/node_modules/nopt
- number-is-nan@1.0.1 node_modules/homebridge/node_modules/number-is-nan
- is-fullwidth-code-point@1.0.0 node_modules/homebridge/node_modules/is-fullwidth-code-point
- oauth-sign@0.8.2 node_modules/homebridge/node_modules/oauth-sign
- object-assign@4.1.1 node_modules/homebridge/node_modules/object-assign
- path-is-absolute@1.0.1 node_modules/homebridge/node_modules/path-is-absolute
- performance-now@0.2.0 node_modules/homebridge/node_modules/performance-now
- process-nextick-args@1.0.7 node_modules/homebridge/node_modules/process-nextick-args
- punycode@1.4.1 node_modules/homebridge/node_modules/punycode
- qs@6.3.2 node_modules/homebridge/node_modules/qs
- minimist@1.2.0 node_modules/homebridge/node_modules/rc/node_modules/minimist
- set-blocking@2.0.0 node_modules/homebridge/node_modules/set-blocking
- signal-exit@3.0.2 node_modules/homebridge/node_modules/signal-exit
- sntp@1.0.9 node_modules/homebridge/node_modules/sntp
- hawk@3.1.3 node_modules/homebridge/node_modules/hawk
- assert-plus@1.0.0 node_modules/homebridge/node_modules/sshpk/node_modules/assert-plus
- string_decoder@0.10.31 node_modules/homebridge/node_modules/string_decoder
- string-width@1.0.2 node_modules/homebridge/node_modules/string-width
- stringstream@0.0.5 node_modules/homebridge/node_modules/stringstream
- strip-json-comments@2.0.1 node_modules/homebridge/node_modules/strip-json-comments
- rc@1.1.7 node_modules/homebridge/node_modules/rc
- ms@0.7.1 node_modules/homebridge/node_modules/tar-pack/node_modules/ms
- debug@2.2.0 node_modules/homebridge/node_modules/tar-pack/node_modules/debug
- tough-cookie@2.3.2 node_modules/homebridge/node_modules/tough-cookie
- tunnel-agent@0.4.3 node_modules/homebridge/node_modules/tunnel-agent
- tweetnacl@0.14.5 node_modules/homebridge/node_modules/tweetnacl
- bcrypt-pbkdf@1.0.1 node_modules/homebridge/node_modules/bcrypt-pbkdf
- sshpk@1.11.0 node_modules/homebridge/node_modules/sshpk
- uid-number@0.0.6 node_modules/homebridge/node_modules/uid-number
- util-deprecate@1.0.2 node_modules/homebridge/node_modules/util-deprecate
- readable-stream@2.2.3 node_modules/homebridge/node_modules/readable-stream
- are-we-there-yet@1.1.2 node_modules/homebridge/node_modules/are-we-there-yet
- readable-stream@2.1.5 node_modules/homebridge/node_modules/tar-pack/node_modules/readable-stream
- uuid@3.0.1 node_modules/homebridge/node_modules/uuid
- verror@1.3.6 node_modules/homebridge/node_modules/verror
- jsprim@1.3.1 node_modules/homebridge/node_modules/jsprim
- http-signature@1.1.1 node_modules/homebridge/node_modules/http-signature
- request@2.80.0 node_modules/homebridge/node_modules/request
- wide-align@1.1.0 node_modules/homebridge/node_modules/wide-align
- gauge@2.7.3 node_modules/homebridge/node_modules/gauge
- npmlog@4.0.2 node_modules/homebridge/node_modules/npmlog
- wrappy@1.0.2 node_modules/homebridge/node_modules/wrappy
- once@1.4.0 node_modules/homebridge/node_modules/once
- inflight@1.0.6 node_modules/homebridge/node_modules/inflight
- glob@7.1.1 node_modules/homebridge/node_modules/glob
- rimraf@2.5.4 node_modules/homebridge/node_modules/rimraf
- fstream@1.0.10 node_modules/homebridge/node_modules/fstream
- fstream-ignore@1.0.5 node_modules/homebridge/node_modules/fstream-ignore
- tar@2.2.1 node_modules/homebridge/node_modules/tar
- once@1.3.3 node_modules/homebridge/node_modules/tar-pack/node_modules/once
- tar-pack@3.3.0 node_modules/homebridge/node_modules/tar-pack
- node-pre-gyp@0.6.33 node_modules/homebridge/node_modules/node-pre-gyp
- bignum@0.12.5 node_modules/homebridge/node_modules/bignum
/opt/node/lib
��������� homebridge@0.4.16 

pi@raspberrypi:~ $ homebridge
bash: homebridge: command not found

PS THANK YOU for all that you do. May take a little longer but it'll get resolved :)

**Comment by NorthernMan54** [comment]
Hum that doesn't look exactly right, what does this command show

node -v
npm -v

I'm expecting some like

v6.9.4 and v3.10.10

**Comment by sdorilas** [comment]
v6.9.5

**Comment by sdorilas** [comment]
3.10.10

**Comment by NorthernMan54** [comment]
Humm installation should look like this

sudo npm install -g homebridge
/home/sgracey/npm/bin/homebridge -> /home/sgracey/npm/lib/node_modules/homebridge/bin/homebridge

> mdns@2.3.3 install /home/sgracey/npm/lib/node_modules/homebridge/node_modules/mdns
> node-gyp rebuild

make: Entering directory '/home/sgracey/npm/lib/node_modules/homebridge/node_modules/mdns/build'
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_sd.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_browse.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_enumerate_domains.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_get_addr_info.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_process_result.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_ref.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_ref_deallocate.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_ref_sock_fd.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_register.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_resolve.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/dns_service_update_record.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/mdns_utils.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/network_interface.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/socket_watcher.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_ref.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_create.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_deallocate.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_set_value.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_get_length.o
  CXX(target) Release/obj.target/dns_sd_bindings/src/txt_record_buffer_to_object.o
  SOLINK_MODULE(target) Release/obj.target/dns_sd_bindings.node
  COPY Release/dns_sd_bindings.node
make: Leaving directory '/home/sgracey/npm/lib/node_modules/homebridge/node_modules/mdns/build'

> curve25519-n@1.1.0 install /home/sgracey/npm/lib/node_modules/homebridge/node_modules/curve25519-n
> node-gyp rebuild

make: Entering directory '/home/sgracey/npm/lib/node_modules/homebridge/node_modules/curve25519-n/build'
  CXX(target) Release/obj.target/curve/node_curve.o
  SOLINK_MODULE(target) Release/obj.target/curve.node
  COPY Release/curve.node
make: Leaving directory '/home/sgracey/npm/lib/node_modules/homebridge/node_modules/curve25519-n/build'

> ed25519@0.0.4 install /home/sgracey/npm/lib/node_modules/homebridge/node_modules/ed25519
> node-gyp rebuild

make: Entering directory '/home/sgracey/npm/lib/node_modules/homebridge/node_modules/ed25519/build'
  CC(target) Release/obj.target/ed25519/src/ed25519/keypair.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sign.o
  CC(target) Release/obj.target/ed25519/src/ed25519/open.o
  CC(target) Release/obj.target/ed25519/src/ed25519/crypto_verify_32.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_double_scalarmult.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_frombytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_scalarmult_base.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_precomp_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p2_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p2_dbl.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_dbl.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_to_p2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_to_cached.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_madd.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_add.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_msub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_sub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p1p1_to_p3.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p1p1_to_p2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_1.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_cmov.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_copy.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_neg.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_add.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_mul.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sq.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sq2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_invert.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_isnegative.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_isnonzero.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_frombytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_pow22523.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sc_reduce.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sc_muladd.o
  CXX(target) Release/obj.target/ed25519/src/ed25519.o
  SOLINK_MODULE(target) Release/obj.target/ed25519.node
  COPY Release/ed25519.node
make: Leaving directory '/home/sgracey/npm/lib/node_modules/homebridge/node_modules/ed25519/build'
/home/sgracey/npm/lib
└─┬ homebridge@0.4.16 
  ├─┬ chalk@1.1.3 
  │ ├── ansi-styles@2.2.1 
  │ ├── escape-string-regexp@1.0.5 
  │ ├─┬ has-ansi@2.0.0 
  │ │ └── ansi-regex@2.1.1 
  │ ├── strip-ansi@3.0.1 
  │ └── supports-color@2.0.0 
  ├─┬ commander@2.8.1 
  │ └── graceful-readlink@1.0.1 
  ├─┬ hap-nodejs@0.4.20 
  │ ├── buffer-shims@1.0.0 
  │ ├─┬ curve25519-n@1.1.0 
  │ │ ├── bindings@1.2.1 
  │ │ └── nan@2.5.1 
  │ ├─┬ debug@2.6.1 
  │ │ └── ms@0.7.2 
  │ ├── ed25519@0.0.4 
  │ ├── fast-srp-hap@1.0.0 
  │ ├── ip@1.1.5 
  │ └─┬ mdns@2.3.3 
  │   └── nan@2.3.5 
  ├─┬ node-persist@0.0.8 
  │ ├── mkdirp@0.3.5 
  │ └── q@1.1.2 
  └── semver@5.0.3 

sgracey@raj:~$

**Comment by sdorilas** [comment]
But it appears everything is installed. What am I missing?

**Comment by NorthernMan54** [comment]
Which node install method did you use?

**Comment by NorthernMan54** [comment]
Your homebridge command is here

 /opt/node/lib/node_modules/homebridge/bin/homebridge

**Comment by sdorilas** [comment]
I tried to use apt-get and I got:

pi@raspberrypi:/opt/node/lib/node_modules/hap-nodejs/node_modules/mdns $ curl -sL https://deb.nodesource.com/setup_7.x | sudo bash -

## Installing the NodeSource Node.js v7.x repo...


## You appear to be running on ARMv6 hardware. Unfortunately this is not currently supported by the NodeSource Linux distributions. Please use the 'linux-armv6l' binary tarballs available directly from nodejs.org for Node.js v4 and later.

**Comment by NorthernMan54** [comment]
What model pi do you have?

**Comment by sdorilas** [comment]
Pi 2.

**Comment by NorthernMan54** [comment]
On my pi 2, what do you have in /etc/apt/sources.list.d/nodesource.list ?

sgracey@howard:/etc/apt/sources.list.d$ cat nodesource.list 
deb https://deb.nodesource.com/node_6.x jessie main
deb-src https://deb.nodesource.com/node_6.x jessie main

**Comment by NorthernMan54** [comment]
On my pi2 I ran this to get everything going

curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash - 
sudo apt-get install -y nodejs
sudo apt-get install libavahi-compat-libdnssd-dev
sudo apt-get install build-essential git vim i2c-tools
sudo npm install -g --unsafe-perm homebridge

**Comment by sdorilas** [comment]
I actually don't have a nodesource.list in that directory. I only have the raspi.list.

**Comment by sdorilas** [comment]
pi@raspberrypi:~ $ curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -

## Installing the NodeSource Node.js v6.x repo...


## You appear to be running on ARMv6 hardware. Unfortunately this is not currently supported by the NodeSource Linux distributions. Please use the 'linux-armv6l' binary tarballs available directly from nodejs.org for Node.js v4 and later.

pi@raspberrypi:~ $ sudo apt-get install -y nodejs
Reading package lists... Done
Building dependency tree       
Reading state information... Done
nodejs is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
pi@raspberrypi:~ $ sudo apt-get install libavahi-compat-libdnssd-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libavahi-compat-libdnssd-dev is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
pi@raspberrypi:~ $ sudo apt-get install build-essential git vim i2c-tools
Reading package lists... Done
Building dependency tree       
Reading state information... Done
build-essential is already the newest version.
git is already the newest version.
git set to manually installed.
i2c-tools is already the newest version.
i2c-tools set to manually installed.
The following extra packages will be installed:
  vim-runtime
Suggested packages:
  ctags vim-doc vim-scripts
The following NEW packages will be installed:
  vim vim-runtime
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 5858 kB of archives.
After this operation, 28.2 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Abort.
pi@raspberrypi:~ $ sudo apt-get install build-essential git vim i2c-tools
Reading package lists... Done
Building dependency tree       
Reading state information... Done
build-essential is already the newest version.
git is already the newest version.
git set to manually installed.
i2c-tools is already the newest version.
i2c-tools set to manually installed.
The following extra packages will be installed:
  vim-runtime
Suggested packages:
  ctags vim-doc vim-scripts
The following NEW packages will be installed:
  vim vim-runtime
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 5858 kB of archives.
After this operation, 28.2 MB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 http://mirrordirector.raspbian.org/raspbian/ jessie/main vim-runtime all 2:7.4.488-7+deb8u2 [5046 kB]
Get:2 http://mirrordirector.raspbian.org/raspbian/ jessie/main vim armhf 2:7.4.488-7+deb8u2 [812 kB]
Fetched 5858 kB in 4s (1406 kB/s)
Can't set locale; make sure $LC_* and $LANG are correct!
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LANG = "en_GB.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
Selecting previously unselected package vim-runtime.
(Reading database ... 112450 files and directories currently installed.)
Preparing to unpack .../vim-runtime_2%3a7.4.488-7+deb8u2_all.deb ...
Adding 'diversion of /usr/share/vim/vim74/doc/help.txt to /usr/share/vim/vim74/doc/help.txt.vim-tiny by vim-runtime'
Adding 'diversion of /usr/share/vim/vim74/doc/tags to /usr/share/vim/vim74/doc/tags.vim-tiny by vim-runtime'
Unpacking vim-runtime (2:7.4.488-7+deb8u2) ...
Selecting previously unselected package vim.
Preparing to unpack .../vim_2%3a7.4.488-7+deb8u2_armhf.deb ...
Unpacking vim (2:7.4.488-7+deb8u2) ...
Processing triggers for man-db (2.7.0.2-5) ...
Setting up vim-runtime (2:7.4.488-7+deb8u2) ...
Processing /usr/share/vim/addons/doc
Setting up vim (2:7.4.488-7+deb8u2) ...
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/vim (vim) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/vimdiff (vimdiff) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/rvim (rvim) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/rview (rview) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/vi (vi) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/view (view) in auto mode
update-alternatives: using /usr/bin/vim.basic to provide /usr/bin/ex (ex) in auto mode

**Comment by NorthernMan54** [comment]
I went back and looked at my PI 2, and it reports ARMv7 not ARMv6 like yours, somethings is not right here.

If you run this, and lookup the result in this table http://elinux.org/RPi_HardwareHistory it will give you the correct model.

cat /proc/cpuinfo | grep 'Revision' | awk '{print $3}' | sed 's/^1000//'

My RPI2, gives a21041, and my RPI3 gives a22082

I think you have an RPI A or B, and you will need to follow the manual installation method for node here http://blog.wia.io/installing-node-js-v4-0-0-on-a-raspberry-pi/ and use a more recent version of node from here https://nodejs.org/dist/latest-v6.x/

Before you get started, you will need to remove the existing version of node with this command

sudo apt-get remove nodejs

And afterwards you should be able to install homebridge with this command

sudo npm install -g --unsafe-perm homebridge

If you run into issues, post the commands you used I will take a look.

**Comment by sdorilas** [comment]
This is my version:

0010	Q3 2014	B+	1.0	512 MB

**Comment by NorthernMan54** [comment]
That's RPI Model B+, and needs to follow the manual installation method

**Comment by sdorilas** [comment]
Pleas pardon my ignorance here: following the guide is going to have me install arm61 just like I had before, right?

It says to run:

wget https://nodejs.org/dist/v4.0.0/node-v4.0.0-linux-armv6l.tar.gz tar -xvf node-v4.0.0-linux-armv6l.tar.gz cd node-v4.0.0-linux-armv6l

**Comment by NorthernMan54** [comment]
Nope, that's an very old version of node and won't work

You need to use a recent version

wget https://nodejs.org/dist/latest-v6.x/node-v6.10.0-linux-armv6l.tar.gz 
tar -xvf node-v6.10.0-linux-armv6l.tar.gz 
cd node-v6.10.0-linux-armv6l
sudo cp -R * /usr/local/

Do you want to switch to slack for this, it would be faster going back and forth?
( https://homebridgeteam.slack.com/messages/homebridge/ )

**Comment by sdorilas** [comment]
We can use slack. Will I need an invite?

**Comment by NorthernMan54** [comment]
The invite is on this page

http://homebridge-slackin.herokuapp.com

**Comment by NorthernMan54** [comment]
I'm logged in and watching #homebridge
